src/types/resource/meta_item.rs

```diff
@@ -34,20 +35,32 @@ struct MetaItemPreviewLegacy {
     r#type: String,
     #[serde(default)]
     name: String,
-    poster: Option<String>,
-    background: Option<String>,
-    logo: Option<String>,
+    #[serde(default)]
+    #[serde_as(deserialize_as = "DefaultOnNull<NoneAsEmptyString>")]
+    poster: Option<Url>,
+    #[serde(default)]
+    #[serde_as(deserialize_as = "DefaultOnNull<NoneAsEmptyString>")]
+    background: Option<Url>,
+    #[serde(default)]
+    #[serde_as(deserialize_as = "DefaultOnNull<NoneAsEmptyString>")]
+    logo: Option<Url>,
     description: Option<String>,
     release_info: Option<String>,
     runtime: Option<String>,
     released: Option<DateTime<Utc>>,
     #[serde(default)]
     poster_shape: PosterShape,
+    #[serde(
+        default,
+        deserialize_with = "deserialize_number_or_text_to_some_string"
+    )]
     imdb_rating: Option<String>,
```

Comment:
can you try to use something like `#[serde_as(deserialize_as = "Option<NumberOrString>")]`
where `NumberOrString` is a struct that implemented [DeserializeAs](https://docs.rs/serde_with/latest/serde_with/trait.DeserializeAs.html) similar to [this](https://github.com/Stremio/stremio-core/blob/catalog-pagination/src/types/serde_as_ext.rs#L14-L31)